src/test/testutil.cpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2009-2016 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "testutil.h"
+
+#ifdef WIN32
+#include <shlobj.h>
+#endif
+
+#include "fs.h"
+
+fs::path GetTempPath() {
+#if BOOST_FILESYSTEM_VERSION == 3
+    return fs::temp_directory_path();
```

Comment:
The version check is unnecessary. I suggest to revert https://github.com/zcash/zcash/pull/4805/commits/dafc0d35eb8f30a925564f8601defa639eddc298, delete `GetTempPath()` from `util.cpp`, and change all uses (there are only three of them in the whole codebase!) to `fs::temp_directory_path()`.

Note that we already use `fs::temp_directory_path()` directly in several places (search for it in this PR). So whatever abstraction `GetTempPath()` was attempting at the time it was added, it's completely redundant now.

Reply:
I'll do this in a new commit, as otherwise the merge conflict I was resolving by the inclusion of those changes will need to be re-handled.